builtin/providers/google/resource_sql_database_instance.go

```diff
@@ -579,6 +603,24 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 		_settings["crash_safe_replication"] = settings.CrashSafeReplicationEnabled
 	}
 
+	if v, ok := _settings["disk_autoresize"]; ok && v != nil {
+		if v.(bool) {
+			_settings["disk_autoresize"] = settings.StorageAutoResize
+		}
+	}
+
+	if v, ok := _settings["disk_size"]; ok && v != nil {
+		if v.(int) > 0 && settings.DataDiskSizeGb < int64(v.(int)) {
```

Comment:
My original version of this (which lives in https://github.com/BashtonLtd/terraform-gcebashtonsql) I didn't have these checks. After some of our instances resized I found the read was making a plan that wished to shrink the database. On a test database I found that if you applied this change the API would reject the request with a 400. This seems to match my understanding of the documentation that the auto resize feature will never shrink the database.

At the time I wrote this I didn't see any other way to prevent the apply. Is there a way to update the state file without necessitating a failing apply?